subprocess/subtree_oom_darwin.go

```diff
@@ -0,0 +1,77 @@
+// +build darwin
+
+package subprocess
+
+import (
+	"bufio"
+	"context"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+func (o *OOMTracker) Clear(ctx context.Context) error {
+	var err error
+	o.IsSudo, err = isSudo(ctx)
+	if err != nil {
+		return errors.Wrap(err, "error checking sudo")
+	}
+
+	if o.IsSudo {
+		return errors.Wrap(exec.Command("sudo", "log", "erase", "--all").Run(), "error clearing log")
+	}
+
+	return errors.Wrap(exec.Command("log", "erase", "--all").Run(), "error clearing log")
+}
+
+func (o *OOMTracker) Check(ctx context.Context) error {
+	wasOOMKilled, pids, err := analyzeLogs(ctx, o.IsSudo)
+	if err != nil {
+		return errors.Wrap(err, "error searching log")
+	}
+	o.WasOOMKilled = wasOOMKilled
+	o.Pids = pids
+	return nil
+}
+
+func analyzeLogs(ctx context.Context, isSudo bool) (bool, []int, error) {
+	var cmd *exec.Cmd
+	wasOOMKilled := false
+	errs := make(chan error)
+
+	if isSudo {
+		cmd = exec.CommandContext(ctx, "sudo", "log", "show")
+	} else {
+		cmd = exec.CommandContext(ctx, "log", "show")
+	}
+	cmdReader, err := cmd.StdoutPipe()
+	if err != nil {
+		return false, []int{}, errors.Wrap(err, "error creating StdoutPipe for log command")
+	}
+
+	scanner := bufio.NewScanner(cmdReader)
+	err = cmd.Start()
+	if err != nil {
+		return false, []int{}, errors.Wrap(err, "Error starting log command")
+	}
+
+	go func() {
+		errs <- cmd.Wait()
+	}()
+
+	pids := []int{}
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.Contains(line, "swap") {
```

Comment:
Is there a longer string than "swap" to look for? I'm concerned "swap" might appear in cases which are no OOM kills. My logs contain, for example,
```
2019-01-17 17:20:22.362391-0500 0x329cb3   Default     0x0                  41     0    UserEventAgent: (com.apple.cts) [com.apple.xpc.activity:All] Unregistered XPC Activity: com.apple.mds.swapIndex.2
2019-01-17 17:20:22.364927-0500 0x329cb3   Default     0x0                  41     0    UserEventAgent: (com.apple.cts) [com.apple.xpc.activity:All] Unregistered XPC Activity: com.apple.mds.swapIndex.3
2019-01-17 17:20:22.366571-0500 0x329cb3   Default     0x0                  41     0    UserEventAgent: (com.apple.cts) [com.apple.xpc.activity:All] Unregistered XPC Activity: com.apple.mds.swapIndex.1
2019-01-17 17:20:22.368761-0500 0x329cb3   Default     0x0                  41     0    UserEventAgent: (com.apple.cts) [com.apple.xpc.activity:All] Unregistered XPC Activity: com.apple.mds.swapIndex.0
```

Reply:
I was pulling swap from the comment in the [link](https://jira.mongodb.org/browse/BUILD-6184?focusedCommentId=2023577&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2023577) that Max mentioned. There they search swap but the messages they're referencing is of the format
`2018-10-03 22:05:39.463368+0000 0x16b Default 0x0 0 kernel: low swap: killing largest compressed process with pid 30613 (mongod) and size 1 MB` 
So maybe "low swap" would be better? I don't know a lot about these log messages but if we aren't sure I can ask Brian McCarthy.

Reply:
I think "low swap" makes sense.